### ISSUE TITLE
UC Cascade parameters do not work on Internet Explorer

### ISSUE THREAD

**Issue by imoutsatsos** [biouno/uno-choice-plugin#72]
Who would have thought to test this one out!

Here is our test server example on Chrome:
![image](https://cloud.githubusercontent.com/assets/3408226/5881438/d1b00770-a30e-11e4-860e-6b2466ac8d53.png)

and here it is on Internet Explorer
![image](https://cloud.githubusercontent.com/assets/3408226/5881445/dfcc8fa4-a30e-11e4-9f7b-1ba8f61bce63.png)

Major issue for me as my users almost exclusively use IE v9.0
![image](https://cloud.githubusercontent.com/assets/3408226/5881478/1cd95ac6-a30f-11e4-873b-d8eba93a5fc0.png)

**Comment by kinow** [comment]
Well, I **think** I have an old Windows box somewhere with a recent version. Give me some time and I'll give it a try. Should be a matter of using different JavaScript methods I suspect - even though we are using JQuery, sometimes I call DOM methods that may be different depending on your browser

**Comment by imoutsatsos** [comment]
Thanks @kinow !! On Chrome everything seems to work great!
In fact I only realized this after I did my 'production roll-out'. Thanks God the weekend will buy me sometime to decide whether to roll back to the pre v0.22 version of Uno-Choice which seems to work as expected on IE. 
As a heads up, it basically looks like the parameters are not passed to the groovy script

**Comment by kinow** [comment]
Let's see if I can get it fixed by Sunday. Maybe you won't have to roll back your environment at all :-)

Thanks for the heads up on the parameters. Guess it is not being able to retrieve the values from the parameters' HTML elements (which it uses DOM methods sometimes).

**Comment by imoutsatsos** [comment]
Thank you @kinow !! Much appreciated! I hope you can work out a solution. These browser specific issues are such a hassle!

**Comment by kinow** [comment]
Hi @imoutsatsos... I tried to reproduce the issue with no success. It worked in a brand new VM running in VirtualBox. 

![001](https://cloud.githubusercontent.com/assets/304786/5889490/501c54a8-a411-11e4-9d25-cb26382a070a.png)

It's been a while since the last time I used Windows (even more time since I used IE), but if I recall correctly, there is some kind of mechanism to enable/disable script compatibility... Screen mode? Something like that in IE... Are you aware of such feature? 

Maybe trying that...

**Comment by imoutsatsos** [comment]
Well I think this is good news @kinow ! I also just tried it with IE v11 running on my Windows 8 at home and it seems to work fine. I'm wondering if somehow the security settings on our corporate IE 9 prevent some javascript actions. Still strange that the older version worked (and still does). Thanks for testing. I will review the issue and script compatibility some more and try to get some ideas as to why it's happening. 

Have a good weekend Bruno! We had a big snowstorm here today. Shoveling snow all day!!

**Comment by kinow** [comment]
Hmmm, the files loaded by the script should all be hosted in Jenkins. But it could be preventing some DOM manipulation maybe. Not sure how strict IE can be with security and scripts.

Thanks Ioannis! Have a great weekend too and I hope the weather will get better! That's a good weather for programming and working on nerd things too, by the way :-D

**Comment by imoutsatsos** [comment]
The issue is reproducible at work using IE9 but i'm currently waiting to upgrade to IE11. Once this is done I will re-try and let you know . For the time-being we can assign this issue a low priority!

**Comment by Scorpio99** [comment]
Me too, i am facing the same issue with IE9. Is the problem fixed?

**Comment by kinow** [comment]
Hi @Scorpio99, I couldn't reproduce the problem in IE9 with a modern.ie virtual machine. If we can find a way to reproduce the issue, it must be quite easy to fix the issue. Do you have more details on your environment configuration that may help us to reproduce it?

**Comment by Scorpio99** [comment]
What details do you need, i am using IE9 with windows 7, 64bit and at work. So may be there are some settings for IE at work which they block, so i am not sure which settings to change on. Tried a few but there was no success.

**Comment by kinow** [comment]
Job configuration (the more details the merrier), Jenkins logs and JavaScript console messages, and OS configuration. Ideally, trying to isolate the problem, i.e. finding the simplest job configuration to reproduce the issue. Thanks!

**Comment by pdirksen** [comment]
I've got the same problem using Internet Explorer **8**.
If you fix this problem using **IE8** that might be solving the issue in IE9.

**Comment by imoutsatsos** [comment]
We have started migrating to IE v11 in my company, and the issue seems resolved. Now, IEv11 works just as good as Chrome with the Uno-Choice plugin.

**Comment by kinow** [comment]
Thanks for the feedback @imoutsatsos 

@pdirksen unless there is something easy that we can do, I'd prefer to spend the development cycles that we have to release 1.0 and fix other issues - like adding support to the security plug-in.

If you have a pull request or patch that fixes the issue we can work on it during the next Mozilla Science Global Sprint, otherwise I think it may be better to close the issue and document that the plug-in is not supported by IE versions prior to 11.

**Comment by pdirksen** [comment]
@kinow Yep, I guess there is something better to do (like adding a password field) than spending time building workarounds for ancient browsers.

**Comment by kinow** [comment]
Exactly @pdirksen, unless it is something really simple or someone else can submit a pull request :-)

**Comment by kinow** [comment]
Closing as we have released 1.0. Please feel free to re-open an issue in https://issues.jenkins-ci.org/ with component name active-choices-plugin in case the issue persists. Thanks!
